build_tools/ci/cpu_comparison/input_generator.py

```diff
@@ -64,6 +64,16 @@ def convert_f32_to_bf16(float32_value):
     return np.uint16(bf16_int_repr)
 
 
+def convert_bf16_to_f32(bfloat16_value):
+    """
+    IEEE bfloat16 to float32. See docstring of convert_f32_to_bf16 for a
+    bit of info on the mantissa/exponent manipulation.
+    """
+    v0 = np.uint32(bfloat16_value) << 16
+    float32_value = np.frombuffer(v0.tobytes(), dtype=np.float32)[0]
+    return float32_value
```

Comment:
wouldn't it make more sense to cast to uint16? won't this foil packing?

Reply:
So, bfloat16_value _is_ a uint16. 

Are you suggesting 

v0 = np.uint16(bfloat16_value) << 16 

? That basically wipes clean all the interesting 16 bits. 

I speak confidently because there's a test in this PR which passes ..

Reply:
Oh, I didn't add the test file lol

Reply:
Test file added

Reply:
i'm saying i don't get why you're using 32 bits for a 16 bit value - i'm assuming that "native" bf16 is packed two (or 4) to a machine word (on the host) where stuff bf16 into fp32 spaces out the values. 

>I speak confidently because there's a test in this PR which passes ..

I don't know any better ... i'm just asking for personal edification

Reply:
From bf16 to fp32:


from:   [SEEEEEEEEMMMMMMM]
to:       [SEEEEEEEEMMMMMMMMMMMMMMMMMMMMMMM]

where the extra mantissa bits are zeros, so we want:
to:       [SEEEEEEEEMMMMMMM0000000000000000]


We start with the following  (which is stored as uint16, bfloat16_value **_is_** uint16, maybe that's what's confusing here)  
[SEEEEEEEEMMMMMMM]

and cast it to uint32:
[0000000000000000SEEEEEEEEMMMMMMM]

and then bitshift it to
[SEEEEEEEEMMMMMMM0000000000000000]

which is what we want.

Reply:
Changing `uint32` to `uint16` seems to have solved the issue with [numerics comparison reported here](https://github.com/nod-ai/iree-amd-aie/pull/829#issuecomment-2395994235).